homeassistant/components/onewire/sensor.py

```diff
@@ -30,33 +31,127 @@
+# These can only be read by OWFS.  Currently this driver only supports them
+# via owserver (network protocol)
+
+HOBBYBOARD_EF = {
+    "HobbyBoards_EF": {
+        "humidity": "humidity/humidity_corrected",
+        "humidity_raw": "humidity/humidity_raw",
+        "temperature": "humidity/temperature",
+    },
+    "HB_MOISTURE_METER": {
+        "moisture_0": "moisture/sensor.0",
+        "moisture_1": "moisture/sensor.1",
+        "moisture_2": "moisture/sensor.2",
+        "moisture_3": "moisture/sensor.3",
+    },
 }
 
 SENSOR_TYPES = {
     "temperature": ["temperature", TEMP_CELSIUS],
     "humidity": ["humidity", "%"],
+    "humidity_raw": ["humidity", "%"],
     "pressure": ["pressure", "mb"],
     "illuminance": ["illuminance", "lux"],
+    "wetness_0": ["wetness", "%"],
+    "wetness_1": ["wetness", "%"],
+    "wetness_2": ["wetness", "%"],
+    "wetness_3": ["wetness", "%"],
+    "moisture_0": ["moisture", "cb"],
+    "moisture_1": ["moisture", "cb"],
+    "moisture_2": ["moisture", "cb"],
+    "moisture_3": ["moisture", "cb"],
+    "counter_a": ["counter", "count"],
+    "counter_b": ["counter", "count"],
+    "HobbyBoard": ["none", "none"],
 }
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Optional(CONF_NAMES): {cv.string: cv.string},
         vol.Optional(CONF_MOUNT_DIR, default=DEFAULT_MOUNT_DIR): cv.string,
+        vol.Optional(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=4304): cv.port,
     }
 )
 
 
+def hb_info_from_type(dev_type="std"):
+    """Return the proper info array for the device type."""
+    if "std" in dev_type:
+        return DEVICE_SENSORS
+    if "HobbyBoard" in dev_type:
+        return HOBBYBOARD_EF
+
+
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the one wire Sensors."""
-    base_dir = config.get(CONF_MOUNT_DIR)
+    base_dir = config[CONF_MOUNT_DIR]
+    owport = config[CONF_PORT]
+    owhost = config.get(CONF_HOST)
     devs = []
     device_names = {}
     if "names" in config:
         if isinstance(config["names"], dict):
             device_names = config["names"]
 
-    if base_dir == DEFAULT_MOUNT_DIR:
+    # We have an owserver on a remote(or local) host/port
+    # pylint: disable=too-many-nested-blocks
+    if owhost:
+        try:
+            owproxy = protocol.proxy(host=owhost, port=owport)
+            devices = owproxy.dir()
+        except protocol.Error as exc:
+            _LOGGER.error(
+                "Cannot connect to owserver on %s:%d, got: %s", owhost, owport, exc
+            )
+            devices = []
+        for device in devices:
+            _LOGGER.debug("found device: %s", device)
+            family = bytes.decode(owproxy.read(f"{device}family"))
+            dev_type = "std"
+            if "EF" in family:
+                dev_type = "HobbyBoard"
+                family = bytes.decode(owproxy.read(f"{device}type"))
+
+            if family in hb_info_from_type(dev_type):
```

Comment:
Invert this check and put the warning if true and `continue`. Then we can outdent below and possibly remove the pylint disable for nested blocks.